File: BOJ/Recommended1/Bronze/2875/2875.py
# 남1, 여2 , 인턴 k 총 남 m, 여 n
n,m,k = map(int,input().split())
_max = 0
for x in range(0,k+1) :
    _max = max(_max,min((n-x) // 2, (m-k+x)))
print(_max)

File: BOJ/Recommended1/Bronze/2875/2875.swift
let input = Dictionary(uniqueKeysWithValues: zip(["n", "m", "k"], readLine()!.split(separator: " ").map { Int($0)! }))

var _max = 0
for x in 0...input["k"]! {
    _max = max(_max,min((input["n"]!-x) / 2, (input["m"]!-input["k"]!+x)))
}
print(_max)